lib/state/app_state.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:async_redux/async_redux.dart';
+import 'package:counter_async_redux/api/models/pokemon_model.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'app_state.freezed.dart';
+part 'app_state.g.dart';
+
+@freezed
+class AppState with _$AppState {
+  factory AppState({
+    @Default(<Pokemon>[]) @JsonKey(name: 'pokemon') List<Pokemon> pokemon,
+    @JsonKey(name: 'nextPage') String? nextPage,
+    @Default(Wait.empty) @JsonKey(ignore: true, name: 'wait') Wait wait,
+  }) = _AppState;
+
+  factory AppState.init() => AppState(
+        wait: Wait(),
```

Comment:
for a single named parameter, there's no need for the trailing comma. 
We only put trailing commas for:
a. multiple named parameters
b. there is a named parameter mixed in with unnamed parameters